ui/qt-lib/src/Ariadne/UI/Qt/Widgets/Dialogs/InsertPassword.hs

```diff
@@ -0,0 +1,118 @@
+module Ariadne.UI.Qt.Widgets.Dialogs.InsertPassword
+  ( runInsertPassword
+  , InsertPasswordResult(..)
+  ) where
+
+import Control.Lens (makeLensesWith)
+
+import Data.Bits
+import qualified Data.Text as T
+
+import Graphics.UI.Qtah.Core.Types (alignHCenter, alignVCenter)
+import Graphics.UI.Qtah.Signal (connect_)
+import Graphics.UI.Qtah.Widgets.QSizePolicy (QSizePolicyPolicy(..))
+
+import qualified Graphics.UI.Qtah.Core.QObject as QObject
+import qualified Graphics.UI.Qtah.Widgets.QAbstractButton as QAbstractButton
+import qualified Graphics.UI.Qtah.Widgets.QBoxLayout as QBoxLayout
+import qualified Graphics.UI.Qtah.Widgets.QCheckBox as QCheckBox
+import qualified Graphics.UI.Qtah.Widgets.QDialog as QDialog
+import qualified Graphics.UI.Qtah.Widgets.QLabel as QLabel
+import qualified Graphics.UI.Qtah.Widgets.QLayout as QLayout
+import qualified Graphics.UI.Qtah.Widgets.QLineEdit as QLineEdit
+import qualified Graphics.UI.Qtah.Widgets.QPushButton as QPushButton
+import qualified Graphics.UI.Qtah.Widgets.QWidget as QWidget
+
+import Ariadne.UI.Qt.UI
+import Ariadne.UI.Qt.Widgets.Dialogs.Util
+import Ariadne.Util
+
+data InsertPassword = InsertPassword
+    { insertPassword :: QDialog.QDialog
+    , hasPassword :: QCheckBox.QCheckBox
+    , password :: QLineEdit.QLineEdit
+    , passwordWidget :: QWidget.QWidget
+    , confirmButton :: QPushButton.QPushButton
+    }
+
+data InsertPasswordResult
+    = InsertPasswordCanceled
+    | InsertPasswordAccepted (Maybe Text)
+
+makeLensesWith postfixLFields ''InsertPassword
+
+initInsertPassword :: IO InsertPassword
+initInsertPassword = do
+    insertPassword <- QDialog.new
+    QObject.setObjectName insertPassword ("insertPasswordDialog" :: String)
+    let headerText = "Insert Password"
+    QWidget.setWindowTitle insertPassword $ toString headerText
+    QWidget.adjustSize insertPassword
+    layout <- createLayout insertPassword
+
+    header <- QLabel.newWithText . toString $ T.toUpper headerText
+    addHeader layout header
+
+    hasPassword <- createCheckBox layout CheckboxOnRight passwordLabelText
+    QAbstractButton.setChecked hasPassword True
+
+    (passwordWidget, passwordLayout) <- createSubWidget
+    passwordLabel <- QLabel.newWithText ("<b>PASSWORD</b>" :: String)
+    (passwordFieldLayout, password) <- createPasswordField "password"
```

Comment:
Here's a slight problem with this. AFAIU, when creating new wallet, password manager would ask to provide a password and set it as wallet's password. However, in this case we'd like user to enter password twice, as confirmation.